analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -171,6 +172,35 @@ private ProgramState ProcessDebugAssertBoolSymbol(ProgramState state, IOperation
         }
     }
 
+    private static ProgramState[] ProcessEquals(SymbolicContext context, IInvocationOperationWrapper invocation)
+    {
+        if (invocation.TargetMethod.IsStatic && invocation.Arguments.Length == 2
```

Comment:
No, we want any static `Equals`, because `string.Equals(.., ..)` does the same job. The idea was that any static `Equals` with two arguments should have predictable behavior.

Instance methods do not make that much sense, because `Null.Equals(other)` will throw and `NotNull.Equals(Null)` doesn't make too much sense. I've chosen not to waste time on it.